### ISSUE TITLE
Multiline messages not working

### ISSUE THREAD

**Issue by vineetchoudhary** [Ilshidur/action-slack#35]
As per the slack message formatting [guide](https://api.slack.com/docs/message-formatting#multiline_messages), we can insert a newline by including the characters `\n`. But this seems to doesn't work in this plugin because I'm getting `\n` in the message.

**Action** - https://github.com/vineetchoudhary/AppBox-iOSAppsWirelessInstallation/blob/master/.github/main.workflow

```
workflow "Slack - New Issues" {
  resolves = ["Slack Notification for New Issues"]
  on = "issues"
}

action "Slack Notification for New Issues" {
  uses = "Ilshidur/action-slack@6aeb2acb39f91da283faf4c76898a723a03b2264"
  secrets = ["SLACK_WEBHOOK"]
  args = "A new issue has been added. \\nURL - {{ EVENT_PAYLOAD.issue.html_url }} \\nTitle - {{ EVENT_PAYLOAD.issue.title }} \\nBody - {{ EVENT_PAYLOAD.issue.body }}"
}
```

![Screenshot_20190909-214921](https://user-images.githubusercontent.com/8479436/64551585-f7d19f80-d352-11e9-89e6-c6702cd18cb8.jpg)

**Comment by Ilshidur** [comment]
Oooooh, this is really bad. Sorry this mistake happened. Good catch :+1: 

Looks like the `\n` gets escaped (to `\\n`) when serializing the payload sent to the Slack servers.
As you pointed out, this escape character seems to have a special treatment when displaying the messages, so I had to make it "un-escaped".

A fix is pushed : https://github.com/Ilshidur/action-slack/commit/87bcfc90b42731c76791c00248019f0a2ef95fdf, so you can use `Ilshidur/action-slack@87bcfc9`.

Let me know if this fixed your problem so I can close this issue ;-)

**Comment by vineetchoudhary** [comment]
I believe `replace` function will replace the first occurrence of `\n`. So, if a message contains multiple`\n` that will not works.

![Screenshot_20190910-082619__01](https://user-images.githubusercontent.com/8479436/64580800-5293e700-d3a5-11e9-82d6-c8c960651845.jpg)

**Comment by Ilshidur** [comment]
Hello.

You are right. Forgetting to use a `RegExp` was a mistake.
This testifies that this code needs some unit tests :smile: 

I released a fix.

**Comment by vineetchoudhary** [comment]
Current solution seems good by looking at code but it's not working. I don't know why. 🙁

![Screenshot_20190910-143549__01](https://user-images.githubusercontent.com/8479436/64600339-8a685200-d3d8-11e9-8bf5-3c8b5cb8e915.jpg)

**Comment by Ilshidur** [comment]
This just needed the `m` Regexp parameter, allowing multi line replacement.

Here's the final fix : `Ilshidur/action-slack@9273a03`

**Comment by vineetchoudhary** [comment]
Finally. 👍
